### PR TITLE
[bugfix] Added ability to configure a secondary vrf

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: client/cloudSubnet_service.go [hunk 1]
```diff
@@ -31,9 +31,28 @@ func (sm *ServiceManager) CreateCloudSubnet(ip string, cloud_cidr_pool_dn string
...
-		jsonPayload.Array(cloudSubnet.ClassName, "children")
 		jsonPayload.ArrayAppend(zoneCon.Data(), cloudSubnet.ClassName, "children")
 	}
+
+	if cloudRsSubnetToCtx != "" {
+		cloudRsSubnetToCtxJSON := []byte(fmt.Sprintf(`
+		{
+			"cloudRsSubnetToCtx": {
+				"attributes": {
+					"annotation":"orchestrator:terraform",
```
Comment (lhercot) [inline client/cloudSubnet_service.go:42]: don't hardcode it. Do you have access to the parent orchestrator value?

File: client/cloudSubnet_service.go [hunk 2]
```diff
@@ -31,9 +31,28 @@ func (sm *ServiceManager) CreateCloudSubnet(ip string, cloud_cidr_pool_dn string
...
 		jsonPayload.ArrayAppend(zoneCon.Data(), cloudSubnet.ClassName, "children")
 	}
+
+	if cloudRsSubnetToCtx != "" {
+		cloudRsSubnetToCtxJSON := []byte(fmt.Sprintf(`
+		{
+			"cloudRsSubnetToCtx": {
+				"attributes": {
+					"annotation":"orchestrator:terraform",	
+					"dn": "%s/%s/rssubnetToCtx",
```
Comment (lhercot) [inline client/cloudSubnet_service.go:43]: do you need the DN?

File: client/cloudSubnet_service.go [hunk 3]
```diff
@@ -64,33 +82,54 @@ func (sm *ServiceManager) DeleteCloudSubnet(ip string, cloud_cidr_pool_addr stri
...
-			"attributes": {
-				"annotation": "orchestrator:terraform",
-				"dn": "%s/%s/rszoneAttach",
-				"tDn": "%s"
+	if zoneDn != "" {
+		rsZoneAttachJSON := []byte(fmt.Sprintf(`
+		{
+			"cloudRsZoneAttach": {
+				"attributes": {
+					"annotation": "orchestrator:terraform",
```
Comment (lhercot) [inline client/cloudSubnet_service.go:99]: don't hardcode it. Do you have access to the parent orchestrator value?

File: client/cloudSubnet_service.go [hunk 4]
```diff
@@ -64,33 +82,54 @@ func (sm *ServiceManager) DeleteCloudSubnet(ip string, cloud_cidr_pool_addr stri
...
+					"dn": "%s/%s/rszoneAttach",
```
Comment (lhercot) [inline client/cloudSubnet_service.go:100]: do you need the DN?

File: client/cloudSubnet_service.go [hunk 5]
```diff
@@ -64,33 +82,54 @@ func (sm *ServiceManager) DeleteCloudSubnet(ip string, cloud_cidr_pool_addr stri
...
 
-	jsonPayload.Array(cloudSubnet.ClassName, "children")
-	jsonPayload.ArrayAppend(zoneCon.Data(), cloudSubnet.ClassName, "children")
+	if cloudRsSubnetToCtx != "" {
+		cloudRsSubnetToCtxJSON := []byte(fmt.Sprintf(`
+		{
+			"cloudRsSubnetToCtx": {
+				"attributes": {
+					"annotation":"orchestrator:terraform",	
+					"dn": "%s/%s/rssubnetToCtx",
```
Comment (lhercot) [inline client/cloudSubnet_service.go:120]: do you need the DN?

File: client/cloudSubnet_service.go [hunk 6]
```diff
@@ -64,33 +82,54 @@ func (sm *ServiceManager) DeleteCloudSubnet(ip string, cloud_cidr_pool_addr stri
...
 	}
 
-	jsonPayload.Array(cloudSubnet.ClassName, "children")
-	jsonPayload.ArrayAppend(zoneCon.Data(), cloudSubnet.ClassName, "children")
+	if cloudRsSubnetToCtx != "" {
+		cloudRsSubnetToCtxJSON := []byte(fmt.Sprintf(`
+		{
+			"cloudRsSubnetToCtx": {
+				"attributes": {
+					"annotation":"orchestrator:terraform",
```
Comment (lhercot) [inline client/cloudSubnet_service.go:119]: don't hardcode it. Do you have access to the parent orchestrator value?

**Review by lhercot — CHANGES_REQUESTED** [review]

File: client/cloudSubnet_service.go (hunk 1 above)
Comment (shrsr) [reply client/cloudSubnet_service.go:42]: We don't have a definition of the orchestrator in the go client flies. 
"orchestrator:terraform" is used as the annotation in the definition of all the relationship attributes payload.

File: client/cloudSubnet_service.go (hunk 3 above)
Comment (shrsr) [reply client/cloudSubnet_service.go:99]: We don't have a definition of the orchestrator in the go client flies.
"orchestrator:terraform" is used as the annotation in the definition of all the relationship attributes payload.

File: client/cloudSubnet_service.go (hunk 5 above)
Comment (shrsr) [reply client/cloudSubnet_service.go:120]: I ran the code without it and we don't require the DN. I removed it.

File: client/cloudSubnet_service.go (hunk 1 above)
Comment (lhercot) [reply client/cloudSubnet_service.go]: Opened https://github.com/ciscoecosystem/aci-go-client/issues/266 to track this in the future

File: client/cloudSubnet_service.go (hunk 3 above)
Comment (lhercot) [reply client/cloudSubnet_service.go:97]: Opened https://github.com/ciscoecosystem/aci-go-client/issues/266 to track this in the future

File: client/cloudSubnet_service.go (hunk 6 above)
Comment (lhercot) [reply client/cloudSubnet_service.go:116]: Opened https://github.com/ciscoecosystem/aci-go-client/issues/266 to track this in the future

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**